src/storage/index.rs

```diff
@@ -0,0 +1,262 @@
+                };
+                index_arr.0.push(index_content);
+                file.seek(SeekFrom::Current(bytes_to_slide as i64));
+                row = row + 1;
+            },
+            Err(e) => {
+                return Err(IndexErr::BuildStringIndexTableError);
+            }
+        };
+    }
+
+    index_arr.0.sort_unstable_by(|a, b| a.key_value.cmp(&b.key_value));
+    Ok(())
+}
+
+pub fn write_string_index_table(table_meta: &TableMeta, index_arr: &mut IndexString) -> Result<(), IndexErr> {
+    let table_index_name = table_meta.table_name.clone() + "index";
+    let mut file_write = File::create(table_index_name).unwrap();
+    for i in 0..index_arr.0.len() {
+        let row_temp = unsafe{mem::transmute::<u32, [u8; 4]>(index_arr.0[i].row)};
+        file_write.write(&row_temp);
+        file_write.write(&index_arr.0[i].key_value);
+    }
+    Ok(())
+}
+
+pub fn read_string_index_table(table_meta: &TableMeta, index_arr: &mut IndexString) -> Result<(), IndexErr> {
+    let table_index_name = table_meta.table_name.clone() + "index";
+    let mut file = File::open(table_index_name).unwrap();
+    let mut buffer_row = [0; 4];
+    let mut buffer_key = vec![0; table_meta.key_bytes as usize];
+    loop {
+        let bytes_read = match file.read(&mut buffer_row) {
+            Ok(0) => break, // end-of-file
+            Ok(n) => {
+                unsafe {
+                    let temp_row = mem::transmute::<[u8; 4], u32>(buffer_row);
+                    file.read(&mut buffer_key);
+                    let mut index_content = IndexDataStructureString {
+                        row: temp_row,
+                        key_value: buffer_key.clone(),
+                    };
+                    index_arr.0.push(index_content);
+                }
+            },
+            Err(e) => {
+                return Err(IndexErr::ReadStringIndexTableError);
+            }
+        };
+    }
+    Ok(())
+}
+
+pub fn insert_string_index_table(insert_value: IndexDataStructureString, index_arr: &mut IndexString) -> Result<(), IndexErr> {
+    if(index_arr.0.is_empty()) {
+        index_arr.0.push(insert_value);
+    } else {
+        let mut target = 0;
+        for i in 0..index_arr.0.len() {
+            if(insert_value.key_value <= index_arr.0[i].key_value) {
+                target = i;
+                break;
+            }
+        }
+        if(target == 0) {
+            index_arr.0.insert(target, insert_value);
+        } else {
+            index_arr.0.insert(target-1, insert_value);
+        }
+    }
+    Ok(())
+}
+
+#[test]
+pub fn test_construct_index() {
+    let table_meta = TableMeta {
+        table_name: String::from("1.in"),
+        key_type: String::from("Int"),
+        key_offet: 0,
+        key_bytes: 4,
+        row_bytes: 4,
+    };
+    let mut index_arr = IndexInt(vec![]);
+    build_int_index_table(&table_meta, &mut index_arr);
+    write_int_index_table(&table_meta, &mut index_arr);
+}
+
+#[test]
```

Comment:
Put the tests in `mod test` to avoid compile them in build time. 
```
#[cfg(test)]
mod tests {
  use super::*;
  #[test] 
  fn test_construct_index() {
  }
}
```